internal/v3/translate/translator.go

```diff
@@ -69,16 +63,18 @@ type APIExporter[T any] interface {
 //	  v20250312:
 //
 // In the above case crdVersion is "v1" and majorVersion is "v20250312".
-func NewTranslator(crd *apiextensionsv1.CustomResourceDefinition, crdVersion string, majorVersion string, deps DependencyRepo) *Translator {
-	return &Translator{
-		crd:          CRDInfo{definition: crd, version: crdVersion},
-		majorVersion: MajorVersion{version: majorVersion},
-		deps:         deps,
+// A translator is an immutable configuration object, it can be safely shared
+// across threads
```

Comment:
this description reads like a description of the struct not the method, hence it should be on the struct. The godoc here should describe what New... does

Reply:
moved to the struct